llama_stack/distribution/routers/routing_tables.py

```diff
@@ -190,3 +206,23 @@ async def register_memory_bank(
         self, memory_bank: MemoryBankDefWithProvider
     ) -> None:
         await self.register_object(memory_bank)
+
+
+class DatasetsRoutingTable(CommonRoutingTableImpl, Datasets):
+    async def list_datasets(self) -> List[DatasetDefWithProvider]:
+        objects = []
+        for objs in self.registry.values():
+            objects.extend(objs)
+        return objects
+
+    async def get_dataset(
+        self, dataset_identifier: str
+    ) -> Optional[ModelDefWithProvider]:
+        return self.get_object_by_identifier(identifier)
+
+    async def register_dataset(self, dataset_def: DatasetDefWithProvider) -> None:
+        await self.register_object(dataset_def)
+
+    async def delete_dataset(self, dataset_identifier: str) -> None:
```

Comment:
does our API have delete also? (fwiw I think we should have delete everywhere in all our object APIs and we should just rename register as create also -- something for later)

Reply:
Our datasets API also have delete, but I'll delete `delete` for now to be consistent with other objects APIs.